packages/next/server/post-process.ts

```diff
@@ -72,12 +72,13 @@ async function processHTML(
     return
   }
 
-  for (let i = 0; i < middlewareRegistry.length; i++) {
-    let middleware = middlewareRegistry[i]
-    if (!middleware.condition || middleware.condition(options)) {
-      await callMiddleWare(middlewareRegistry[i].middleware)
-    }
-  }
+  await Promise.allSettled(
+    middlewareRegistry.map(
+      (middleWare) =>
+        (!middleware.condition || middleware.condition(options)) &&
+        callMiddleWare(middleWare?.middleware)
+    )
+  );
```

Comment:
The middlewares should be executed in order where one is dependent on another.

Reply:
> Middlewares should be executed in order since middleware is dependent on one another.

Ok, sorry :(